components/org.wso2.carbon.appmgt.gateway/src/main/java/org/wso2/carbon/appmgt/gateway/internal/AppManagerGatewayComponent.java

```diff
@@ -93,7 +92,7 @@ protected void setAPIManagerConfigurationService(AppManagerConfigurationService
             log.debug("Gateway manager configuration service bound to the WebApp host objects");
         }
         configuration = amcService.getAPIManagerConfiguration();
-        ServiceReferenceHolder.getInstance().setAPIManagerConfigurationService(amcService);
+        org.wso2.carbon.appmgt.impl.internal.ServiceReferenceHolder.getInstance().setAPIManagerConfigurationService(amcService);
```

Comment:
Service Reference holder should remain as it was "internal". Wire the required OSGI services like done on other ServiceReferenceHolder classes.